lib/compat/wordpress-6.8/class-gutenberg-rest-post-search-handler.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+add_filter( 'wp_rest_search_handlers', 'gutenberg_rest_post_search_handler' );
+
+class Gutenberg_REST_Post_Search_Handler extends WP_REST_Post_Search_Handler {
+	public function prepare_item( $id, $fields ) {
+		$data = parent::prepare_item( $id, $fields );
+		if ( (int) get_option( 'page_on_front' ) === $id ) {
```

Comment:
Any concerns with calling `get_option` on every item?

Reply:
My other concern, of course, is whether this — adding two new properties to search results that suit our application's needs — is a bad precedent to set. Are these properties useful enough to eventually land in core's actual search handler? Alternatively, could we pass a flag in our search request as a condition to get these properties back?